src/app/routes/cpsAsset/utils/getAdditionalPageData.js

```diff
@@ -75,16 +129,31 @@ const pageTypeUrls = async (
 
 const validateResponse = ({ status, json }, name) => {
   if (status === 200 && !isEmpty(json)) {
+    // 004_brasil_recommendations_experiment
     return { [name]: json };
   }
 
   return null;
 };
 
-const fetchUrl = ({ name, path, ...loggerArgs }) =>
-  fetchPageData({ path, timeout: SECONDARY_DATA_TIMEOUT, ...loggerArgs })
-    .then(response => validateResponse(response, name))
-    .catch(noop);
+const fetchUrl = async ({ name, path, attachAgent, ...loggerArgs }) => {
+  // 004_brasil_recommendations_experiment
+  try {
+    const agent = attachAgent ? await getAgent() : null;
+
+    return fetchPageData({
+      path,
+      timeout: SECONDARY_DATA_TIMEOUT,
+      agent,
+      ...loggerArgs,
+    })
+      .then(response => validateResponse(response, name))
+      .catch(noop);
+  } catch (error) {
+    console.log(error);
```

Comment:
Can we use the logger here so any errors make it to our logging on the server

Here's an example: https://github.com/bbc/simorgh/blob/a0fd1cacc7c6accfb38a42ac0af48981c0091343/src/app/routes/utils/fetchPageData/index.js#L51